### PR TITLE
♻️ Move RTC to a service

### PR DESCRIPTION
**Pull request by powerivq**
https://github.com/ampproject/amphtml/pull/31101

It moves the rtc related code into a service.

### CODE REVIEW

File: extensions/amp-a4a/0.1/amp-a4a.js [hunk 1]
```diff
@@ -2361,31 +2362,23 @@ export class AmpA4A extends AMP.BaseElement {
...
-          `${this.element.getAttribute('type')}`
+    if (this.element.getAttribute('rtc-config')) {
+      installRealTimeConfigServiceForDoc(this.getAmpDoc());
+      return this.getBlockRtc_().then((shouldBlock) =>
+        shouldBlock
+          ? undefined
+          : Services.realTimeConfigForDoc(
+              this.getAmpDoc()
+            ).then((realTimeConfig) =>
+              realTimeConfig.maybeExecuteRealTimeConfig(
```
Comment (calebcordry) [inline extensions/amp-a4a/0.1/amp-a4a.js:2373]: it seems like this can be `undefined`?  ie cannot read property maybeExec of undefined...

**Review by calebcordry — CHANGES_REQUESTED** [review]

File: extensions/amp-a4a/0.1/amp-a4a.js (hunk 1 above)
Comment (calebcordry) [reply extensions/amp-a4a/0.1/amp-a4a.js:2373]: never mind, indentation confused me :)

**Review by calebcordry — APPROVED** [review]

**Comment by powerivq** [comment]
Adding @jridgewell for src/service owner approval.

**Review by jridgewell — APPROVED** [review]
Owners approval

**Pull request merged**